File: 2022/01/python/main.py
INPUT_FILE = "../input/input.txt"

list = []
total = 0

# --- Read and parse the input file ---
file = open(INPUT_FILE)
while True:
    line = file.readline()
    if not line:
        break

    line = line.strip()
    if len(line) == 0:
        list.append(total)
        total = 0
    else:
        total += int(line)
file.close()
list.append(total)
list.sort(reverse = True)

# --- Find the Elf carrying the most Calories ---
max = list[0]
print(f"The Elf carrying the most Calories, is carrying {max:,d} Calories.")

# --- Find the top three Elves carrying the most Calories ---
total = sum(list[0:3])
print(f"The top 3 Elves carrying the most Calores, are carrying {total:,d} Calories.")

File: 2022/01/rust/main.rs
use std::fs::File;
use std::io::{ BufRead, BufReader };

const INPUT_FILE:&'static str = "../input/input.txt";

fn main() -> Result<(), std::io::Error> {
    let mut list:Vec<u64> = Vec::new();
    let mut total: u64 = 0;

    // --- Read and parse the input file ---
    let file = File::open(INPUT_FILE)?;
    for line in BufReader::new(file).lines() {
        let line = line.unwrap();
        if line.len() == 0 {
            list.push(total);
            total = 0;
        } else {
            total += line.parse::<u64>().unwrap();
        }
    }
    list.sort();
    list.reverse();

    // --- Find the Elf carrying the most Calories ---
    let max = list[0];
    println!("The Elf carrying the most Calories, is carrying {} Calories.", max);

    // --- Find the top three Elves carrying the most Calories ---
    total = list[0..3].iter().sum();
    println!("The top 3 Elves carrying the most Calores, are carrying {} Calories.", total);
    Ok(())
}
